packages/tools/client-logger/app-insights-logger/src/appInsightsLogger.ts

```diff
@@ -0,0 +1,42 @@
+import { ApplicationInsights, Snippet } from "@microsoft/applicationinsights-web";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+export interface AppInsightsLoggerConfig {
+	appInsightsClient?: ApplicationInsights;
```

Comment:
Is there a reason to ask the consumer for an instantiated AppInsights client instead of always instantiating it ourselves inside the logger? It feels a bit weird. It does have the plus of making this more testable, I'll give it that.

Current logic in the constructor acts differently in each case though; when we build it, we end up calling `.loadAppInsights()` in the end which we don't do if we receive the instance, but then we'd need to document that consumers are responsible for that in that case.

Reply:
1. I adjusted this interface in the latest commit, check it out.
2. I wanted to give consumers flexibility with how they instantiate their client. Before the .loadAppInsights() method, the App Insights instance can be configured using additional methods, event listeners, etc. I can't tell how they'll use it in a complex setup so I opted for flexibility.

Reply:
With that in mind, I think maybe we're better off letting the consumer always instantiate and configure the base client and giving us the instantiated object. For a "simple consumer", who wouldn't do much custom configuration, it'd just be a couple of lines like so, right?

```typescript
const client = new ApplicationInsights({ connectionString: "<value>"});
client.loadAppInsights();

const logger = new FluidAppInsightsLogger(client);
```